Add options to keep output and/or execution counts

jpeacock29 opened the pull request:
I developed this for my own use case where I wanted to version control the source and its outputs, but not the execution count and other miscellaneous metadata. These two options give fine grained control of nbstripout's behaviour.

kynan commented on nbstripout.py:
```diff
@@ -240,6 +258,10 @@ def main():
                       help='Check if nbstripout is installed in current repository')
     task.add_argument('--status', action='store_true',
                       help='Print status of nbstripout installation in current repository and configuration summary if installed')
+    parser.add_argument('-c', '--count', action='store_true', help="""
```
I'd prefer to only provide the long option and call it `--keep-count`

kynan commented on nbstripout.py:
```diff
@@ -240,6 +258,10 @@ def main():
                       help='Print status of nbstripout installation in current repository and configuration summary if installed')
+    parser.add_argument('-c', '--count', action='store_true', help="""
+                        Do not strip the execution count/prompt number""")
+    parser.add_argument('-o', '--output', action='store_true', help="""
```
I'd prefer to only provide the long option and call it `--keep-output`.

kynan commented on nbstripout.py:
```diff
@@ -240,6 +258,10 @@ def main():
     task.add_argument('--status', action='store_true',
                       help='Print status of nbstripout installation in current repository and configuration summary if installed')
+    parser.add_argument('-c', '--count', action='store_true', help="""
+                        Do not strip the execution count/prompt number""")
```
For consistency, could you put the help option on the 2nd line and enclose the help text in `''`

kynan commented on nbstripout.py:
```diff
@@ -240,6 +258,10 @@ def main():
+    parser.add_argument('-c', '--count', action='store_true', help="""
+                        Do not strip the execution count/prompt number""")
+    parser.add_argument('-o', '--output', action='store_true', help="""
+                        Do not strip the output""")
```
For consistency, could you put the help option on the 2nd line and enclose the help text in `''`

kynan commented on nbstripout.py:
```diff
@@ -129,20 +129,38 @@ def _cells(nb):
+                cell['outputs'] = []
+            else:
+                for output in cell['outputs']:
+                    if 'execution_count' in output:
```
Should this also be conditional on `keep_count`?

jpeacock29 replied:
Could catch; fixed.

kynan commented:
Could you please rebase on master and resolve the conflict?

kynan commented:
If you're interest you could also add the changes to address #52

jpeacock29 commented:
I added a fix for #52 , but I'm not 100% on the intended functionality, so take a look.

kynan commented:
Thanks. Could you please fix the failing test? In fact, adding a new test for this would be great!

jpeacock29 commented:
1. Failing test is fixed (changed to reflect new correct output per #52).
2. Fixed a bug in handling keep_output/init cells.
3. Added 3 tests for --keep-output, --keep-count and the combination.

kynan commented:
Excellent, thank you very much for your contribution!

jpeacock29 commented:
My pleasure, thanks for merging! I was also considering adding some documentation to the README.rst, since users might only read the docs on GitHub rather than `nbstripout --help`. What are your thoughts?

kynan commented:
Yes indeed, please do!
